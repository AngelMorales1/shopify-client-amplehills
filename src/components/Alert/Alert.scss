@import 'settings';

.Alert {
  height: $alert-height-mobile;

  @media ($medium-up) {
    height: $alert-height-desktop;
  }

  &__text {
    font-size: $font-1;

    @media ($medium-up) {
      font-size: $font-2;
    }
  }

  &__linked-text {
    margin-left: spacing(1);
    text-decoration: none;
  }
}
