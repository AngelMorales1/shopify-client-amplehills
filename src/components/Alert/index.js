import React from 'react';
import cx from 'classnames';
import get from 'utils/get';
import { Button } from 'components/base';

import styles from './Alert.scss';

const Alert = ({ alert }) => {
  const fields = get(alert, 'fields', {});

  return (
    <div
      className={cx(
        styles['Alert'],
        'z-nav w100 fixed l0 t0 bg-madison-blue bold flex flex-row justify-center items-center'
      )}
    >
      <span className={cx(styles['Alert__text'], 'text-white')}>
        {fields.alertCopy}
      </span>
      <Button
        to={fields.linkUrl}
        variant="style-none"
        className={cx(
          styles['Alert__order-now'],
          styles['Alert__text'],
          'text-peach'
        )}
        label={fields.linkCopy}
      />
    </div>
  );
};

export default Alert;
