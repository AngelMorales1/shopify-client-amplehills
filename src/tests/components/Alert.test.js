import React from 'react';
import { shallow } from 'enzyme';

import Alert from 'components/Alert';

it('renders without data', () => {
  const component = shallow(<Alert />);

  expect(component).toMatchSnapshot();
});

it('renders with data', () => {
  const component = shallow(
    <Alert
      alert={{
<<<<<<< HEAD
        alertCopy: 'foo',
        linkCopy: 'foo',
        linkUrl: 'foo',
        title: 'foo'
=======
        fields: {
          alertCopy: 'foo',
          linkCopy: 'foo',
          linkUrl: '/foo',
          title: 'foo'
        },
        sys: {
          contentType: {
            sys: {}
          },
          createdAt: 'foo',
          environment: {
            sys: {}
          },
          id: 'foo',
          locale: 'foo',
          revision: 1,
          space: {
            sys: {}
          },
          type: 'foo',
          updatedAt: 'foo'
        }
>>>>>>> Add Alert component test
      }}
    />
  );

  expect(component).toMatchSnapshot();
});
