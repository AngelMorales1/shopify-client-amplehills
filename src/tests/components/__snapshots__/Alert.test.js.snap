// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Alert
    alert={
        Object {
            "alertCopy": "foo",
            "linkCopy": "foo",
            "linkUrl": "foo",
            "title": "foo",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="text-white"
>
          foo
</span>,
        <a
          className="text-peach"
          href="foo"
          rel="noopener"
>
          foo
</a>,
      ],
      "className": "z-overlay w100 fixed l0 t0 px1 bg-madison-blue bold flex flex-row justify-center items-center flex-wrap transition-fade-in center",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "foo",
          "className": "text-white",
        },
        "ref": null,
        "rendered": "foo",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "foo",
          "className": "text-peach",
          "href": "foo",
          "rel": "noopener",
        },
        "ref": null,
        "rendered": "foo",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="text-white"
>
            foo
</span>,
          <a
            className="text-peach"
            href="foo"
            rel="noopener"
>
            foo
</a>,
        ],
        "className": "z-overlay w100 fixed l0 t0 px1 bg-madison-blue bold flex flex-row justify-center items-center flex-wrap transition-fade-in center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "foo",
            "className": "text-white",
          },
          "ref": null,
          "rendered": "foo",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "foo",
            "className": "text-peach",
            "href": "foo",
            "rel": "noopener",
          },
          "ref": null,
          "rendered": "foo",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Alert
    alert={
        Object {
            "alertCopy": "",
            "linkCopy": "",
            "linkUrl": "",
            "title": "",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="text-white"
/>,
        <a
          className="text-peach"
          href=""
          rel="noopener"
/>,
      ],
      "className": "z-overlay w100 fixed l0 t0 px1 bg-madison-blue bold flex flex-row justify-center items-center flex-wrap transition-fade-in center",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "text-white",
        },
        "ref": null,
        "rendered": "",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "text-peach",
          "href": "",
          "rel": "noopener",
        },
        "ref": null,
        "rendered": "",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="text-white"
/>,
          <a
            className="text-peach"
            href=""
            rel="noopener"
/>,
        ],
        "className": "z-overlay w100 fixed l0 t0 px1 bg-madison-blue bold flex flex-row justify-center items-center flex-wrap transition-fade-in center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "text-white",
          },
          "ref": null,
          "rendered": "",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "text-peach",
            "href": "",
            "rel": "noopener",
          },
          "ref": null,
          "rendered": "",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
